packages/runtime/garbage-collector/src/utils.ts

```diff
@@ -162,11 +162,49 @@ export function concatGarbageCollectionStates(
     return { gcNodes: combinedGCNodes };
 }
 
+/**
+ * Concatenates the given GC datas and returns the concatenated GC data.
+ */
+export function concatGarbageCollectionDatas(gcData1: IGarbageCollectionData, gcData2: IGarbageCollectionData) {
+    const combinedGCData: IGarbageCollectionData = cloneGCData(gcData1);
+    for (const [id, routes] of Object.entries(gcData2.gcNodes)) {
+        if (combinedGCData.gcNodes[id] === undefined) {
+            combinedGCData.gcNodes[id] = Array.from(routes);
+        } else {
+            const combinedRoutes = [ ...routes, ...combinedGCData.gcNodes[id] ];
+            combinedGCData.gcNodes[id] = [ ...new Set(combinedRoutes) ];
```

Comment:
What does [...new Set(combinedRoutes)] do? It seems like it should be equivalent to just doing [...combinedRoutes], no?
Is it about removing duplicates?

Reply:
Yes, it is for removing duplicates.